### ISSUE TITLE
LB: Intermediate Kubernetes node is identified as world

### ISSUE THREAD

**Issue by Cajga** [cilium/cilium#25650]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

We have several nodeports service defined so, we can reach these pods from outside of the cluster. We allow the external host(s) to reach the nodeport on the nodes (we have host firewall enabled in Cilium) but according to audit logs, the traffic would still be blocked. We see the following entry with `hubble observe -f --verdict AUDIT`:
```
May 24 13:07:41.178: 100.80.2.123:8311 (world) -> flux-system/notification-controller-cc887b49d-5hm5l:8080 (ID:9384) policy-verdict:none INGRESS AUDITED (TCP Flags: SYN)
```

The 100.80.2.0/24 subnet (from where the source address is in the logs above) is assigned to the kubernetes node which we used to reach the service through the nodeport:
```
$ kubectl describe nodes hostnameofthenode |grep 100.80.2.0
                    io.cilium.network.ipv4-pod-cidr: 100.80.2.0/24
PodCIDR:                             100.80.2.0/24
PodCIDRs:                            100.80.2.0/24
```

So, seems for the `nodeports service` to work, I would also need to allow this traffic. So, I tried this with the following policy:
```
apiVersion: cilium.io/v2
kind: CiliumNetworkPolicy
metadata:
  name: prometheus-scraping-node-to-flux
  namespace: flux-system
spec:
  endpointSelector: {}
  ingress:
    - fromEntities:
      - host
      - remote-node
```

But it did not help, the audit log showed the same...

So, seems cilium does not recognize the kubernetes node in such cases (using nodeport service) hence the word `world` is used in the AUDIT log entry.

### Cilium Version

```
$ cilium version
cilium-cli: v0.14.3 compiled with go1.20.2 on linux/amd64
cilium image (default): v1.13.2
cilium image (stable): v1.13.2
cilium image (running): v1.13.2
```

### Kernel Version

```
# uname -a
Linux kls011 5.4.0-148-generic #165-Ubuntu SMP Tue Apr 18 08:53:12 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

### Kubernetes Version

```
$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.1", GitCommit:"4c9411232e10168d7b050c49a1b59f6df9d7ea4b", GitTreeState:"clean", BuildDate:"2023-04-14T13:21:19Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.4-eks-597964d", GitCommit:"597964dc3dceb9ce405f028017ae8251929612ec", GitTreeState:"archive", BuildDate:"2023-04-12T12:05:34Z", GoVersion:"go1.19.8", Compiler:"gc", Platform:"linux/amd64"}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Cajga** [comment]
So this source IP is coming from the cilium_host interface on the kubernetes node:
```
# ip addr show cilium_host
9: cilium_host@cilium_net: <BROADCAST,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 12:4f:c6:b4:a8:7a brd ff:ff:ff:ff:ff:ff
    inet 100.80.2.123/32 scope global cilium_host
       valid_lft forever preferred_lft forever
    inet6 fe80::104f:c6ff:feb4:a87a/64 scope link 
       valid_lft forever preferred_lft forever
```

Does these cilium_host IP have a reserved entity?

**Comment by Cajga** [comment]
Interestingly the own IP of the host is not listed as endpoint. This is running inside the cilium-agent on the same host:
```
# cilium endpoint list|grep -i 100.80.2.123
#
```

**Comment by pchaigno** [comment]
This is expected and happens because the packet was SNATed and we can't deduce the identity of the original source IP.

This is usually worked around by using DSR or externalTrafficPolicy=Local to preserve the source IP address.

**Comment by Cajga** [comment]
@pchaigno , thanks a lot again for the explanation and support.

Relevant sources for others:
- https://docs.cilium.io/en/v1.13/network/kubernetes/kubeproxy-free/#client-source-ip-preservation
- https://docs.cilium.io/en/v1.13/network/kubernetes/kubeproxy-free/#direct-server-return-dsr

**Comment by Cajga** [comment]
@pchaigno, these solutions only work with the kubeproxy-free mode. Unfortunately, the distribution that we are using (EKS Anywhere Bare Metal) does not support kube-proxy free installation at the moment (I will follow up with AWS on this if it was supported/safe to disable it after installation) and as such we will need to stick with the kube-proxy installed...

As we have the host firewall mode on, I am already filtering the source IPs from outside the cluster. Would you see any issues allowing the `world` identity to reach the pods? I mean, the `world` is basically filtered at the nodeport access level and no pods will have `world` identity inside the cluster... Am I missing something?

**Comment by pchaigno** [comment]
>  these solutions only work with the kubeproxy-free mode.

externalTrafficPolicy=Local should also work with kube-proxy.

> Would you see any issues allowing the world identity to reach the pods? I mean, the world is basically filtered at the nodeport access level and no pods will have world identity inside the cluster... Am I missing something?

Anything that we can't recognize will have the world identity. That should indeed not include pods from the cluster.

**Comment by turkmenkaan** [comment]
The other issue here is this prevents having firewall rules around the Clustermesh LoadBalancer. If `externalTrafficPolicy = Cluster`, network policies start dropping traffic coming from the other cluster in the Clustermesh. `externalTrafficPolicy = Local` seems to be an option for LoadBalancers with targets on every node (a.k.a. daemonsets), however for workloads like clustermesh-apiserver that is not guaranteed to be on the every node, this leads to potential issues as `externalTrafficPolicy = Local` only directs traffic to pods on the same node as the VIP announcement. 

In summary, it seems like due to this issue, it is not possible to have firewall rules for Cilium clustermesh-apiserver service.
